src/EditorFeatures/Core/Implementation/MetadataAsSource/MetadataAsSourceFileService.cs

```diff
@@ -139,6 +139,15 @@ public async Task<MetadataAsSourceFile> GetGeneratedFileAsync(Project project, I
                         }
                         catch (Exception e) when (FatalError.ReportWithoutCrashUnlessCanceled(e))
                         {
+                            // Insert the exception message at the top of the fallback Metadata As Source document to
+                            // help with diagnosing failures. When the generated source is added to the document, this
+                            // comment will end up at the bottom.
+                            var failureText = e.ToString();
+                            var failureTextAsComment = "// " + failureText.Replace("\n", "\n// ") + Environment.NewLine;
```

Comment:
Sorry one more thing that would be nice: add some text along the lines of "the decompilation failed, and here was the failure". Otherwise people might wonder why there's a random exception at the end of the file.